eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Mgmt/src/Providers/ResourceClientProvider.cs

```diff
@@ -223,14 +227,13 @@ protected override MethodProvider[] BuildMethods()
             return [BuildValidateResourceIdMethod(), .. operationMethods];
         }
 
-        private MethodProvider BuildOperationMethod(InputServiceMethod method, MethodProvider convenienceMethod, bool isAsync, bool isUpdateOnly = false)
+        protected MethodProvider BuildOperationMethod(InputServiceMethod method, MethodProvider convenienceMethod, bool isAsync, bool isUpdateOnly = false)
```

Comment:
Given the complexity of BuildOperationMethod grows, consider to abstract this part to an isolated place such as `ResourceOperationMethodBuilder`.
Then, we can have inheritance properly for the different methods implementation.
Let's add a TODO for now.

Reply:
TODO added. We need a follow-up PR to refactor build operation method related code.